fs/dafsserver/dafsserver.cpp

```diff
@@ -2360,6 +2422,15 @@ IFileDescriptor *verifyMetaInfo(IPropertyTree &actNode, bool authorizedOnly, con
     return fileDesc.getClear();
 }
 
+template<class ActivityClass> IRemoteReadActivity *createConditionalProjectingActivity(IPropertyTree &actNode, IFileDescriptor *fileDesc)
+{
+    Owned<IRemoteReadActivity> activity = new ActivityClass(actNode, fileDesc);
+    if (activity->requiresPostProject())
+        return new CRemoteCompoundReadProjectActivity(actNode, activity);
```

Comment:
What about if input and output are supplied, but identical? This project will not be needed..

Reply:
is there a way to compare IOutputMeta's (or their type info) for equality?

Reply:
not very easily.  I have a related issue open, I'll add a comment to that.